src/connectivity/mod.rs

```diff
@@ -676,13 +737,6 @@ pub fn digraph_core_number(
 /// non-overlapping segment, which is a path instead of a cycle. Each
 /// cycle or path is called a *chain*. For more information, see [Schmidt]_.
 ///
-/// .. note::
-///
-///     The function implicitly assumes that there are no parallel edges
-///     or self loops. It may produce incorrect/unexpected results if the
-///     input graph has self loops or parallel edges. It's also a recursive
-///     implementation and might run out of memory in large graphs.
-///
```

Comment:
No, I reverted this in 392ee24.
But it reminded me that we should use in `chain_decomposition` the iterative dfs-search implementation we added in #456 instead of the recursive petgraph's implementation. I'll push a separate PR to fix this.

Reply:
Done in #552.